.github/workflows/test.yml

```diff
@@ -4,9 +4,11 @@ on:
   push:
     branches:
       - master
+    paths-ignore:
+      - '**.md'
   pull_request:
```

Comment:
we can probably drop pull_request in favor of push without a branch filter

Reply:
The `pull_request` event is required to report checks when a pull request is submitted from a fork. So you'll see this pattern all over:

https://github.com/actions/setup-node/blob/main/.github/workflows/build-test.yml#L3-L12

The push branch filter is necessary to ensure that pull requests opened from branches pushed directly to the repo don't have duplicate checks.

Leaving out `pull_request` for private repos might be fine 99% of the time, but even then it's best to include it. Forking is still useful to allow users with read-only access to propose changes.

Reply:
good to know! i had pull_request pegged as a subset of the push events